mods/cnc/maps/shellmap/shellmap.lua

```diff
@@ -1,3 +1,362 @@
+			a.Stop()
+			a.AttackMove(loc)
+			Trigger.AfterDelay(250, function() StayInLocation(a,loc,dist) end)
+		end
+	end
+end
+MakeNod = function(fact,teams)
+	nod.Cash = nod.Cash + 3000
+	if not fact.IsDead then
+		local teamtype = Utils.Random(teams)
+		fact.Build(teamtype[1], function(team)
+			Utils.Do(team, teamtype[2])
+			MakeNod(fact,teams)
+		end)
+	end
+end
+MakeGDI = function(fact,teamtype)
+	gdi.Cash = gdi.Cash + 3000
+	fact.Build(teamtype, function(team)
+		Utils.Do(team, function(a)
+			a.Stance = "Defend"
+			StayInLocation(a,MCVDeploy.Location-CVec.New(3,7),1)
+		end)
+		Trigger.OnAllKilled(team,function()
+			if not fact.IsDead then
+				MakeGDI(fact,teamtype)
+			end
+		end)
+	end)
+end
+MakeUnitInvincible = function(unit)
+	if not unit.HasProperty("AcceptsUpgrade") then
+		unit.Destroy()
+	end
+	if unit.HasProperty("AcceptsUpgrade") and unit.AcceptsUpgrade("unkillable") then
+		unit.GrantUpgrade("unkillable")
+		if unit.Stance ~= nil then
+			unit.Stance = "Defend"
+		end
+	end
+end
+--Repeated Functions END
+
+--Event Sequence Functions START
+SendCommandos = function()
+	local commandos = Reinforcements.ReinforceWithTransport(gdi, "tran", CommandoTeam, 
+	{ChinEntry.Location, ChinUnload.Location}, {ChinExit.Location})[2]
+	
+	Trigger.OnIdle(commandos[1], function()
+			if(Map.NamedActor("Turret1") ~= nil) then
+				commandos[1].Demolish(Turret1)
+			end
+			if(Map.NamedActor("beachdude1") ~= nil or Map.NamedActor("beachdude2") ~= nil) then
+				commandos[1].AttackMove(CommMove.Location)
+			end
+			if(Map.NamedActor("Turret3") ~= nil) then
+				commandos[1].Demolish(Turret3)
+				commandos[1].Move(CommFin.Location)
+			end
+			commandos[1].Move(CommFin.Location)
+	end)
+
+	Trigger.OnIdle(commandos[2], function()
+			if(Map.NamedActor("Turret2") ~= nil) then
+				commandos[2].Demolish(Turret2)
+			end
+			if(Map.NamedActor("beachdude1") ~= nil or Map.NamedActor("beachdude2") ~= nil) then
+				commandos[2].AttackMove(CommMove.Location)
+			end
+			if(Map.NamedActor("Turret4") ~= nil) then
+				commandos[2].Demolish(Turret4)
+				commandos[2].Move(CommFin.Location)
+			end
+			commandos[2].Move(CommFin.Location)
+	end)
+
+	Trigger.OnKilled(Turret1, function()
+			beachdude1.AttackMove(BeachDudeMove.Location)
+			beachdude2.AttackMove(BeachDudeMove.Location)
+	end)
+
+	Trigger.OnKilled(Turret3, function()
+			commandos[1].Stop()
+			commandos[1].ScriptedMove(CommFin.Location)
+			Trigger.Clear(commandos[1], "OnIdle")
+			Trigger.AfterDelay(25, Beachhead)
+	end)
+
+	Trigger.OnKilled(Turret4, function()
+			commandos[2].Stop()
+			commandos[2].ScriptedMove(CommFin.Location)
+			Trigger.Clear(commandos[2], "OnIdle")
+	end)
+	Trigger.OnAllKilled({beachdude1, beachdude2}, function()
+		local trans = Reinforcements.ReinforceWithTransport(gdi, "tran", nil, {EvacEntry.Location, EvacLand.Location})[1]
+		Trigger.OnIdle(trans, function()
+			if(trans.HasPassengers) then
+				trans.Wait(25)
+				trans.CallFunc(function() PlayMusic() end)
```

Comment:
You already start the music in the `WorldLoaded` function, and here again. Is that on purpose?